### PR TITLE
Add classes for Add, Mul and Pow

### PR DESCRIPTION
**Pull request by rajithv**
Changing the class of addition, multiplication and power expressions to `Add`, `Mul`, and `Pow` to reflect SymEngine classes.

### CODE REVIEW

File: spec/arit_spec.rb [hunk 1]
```diff
@@ -54,22 +54,32 @@
...
     assert(e.to_s == 'x*y') || 'y*x'
+    expect(e).to be_a SymEngine::Mul
+
     e = Integer(2) * x
     assert e.to_s == '2*x'
+    expect(e).to be_a SymEngine::Pow
+
     e = 2 * x
     assert e.to_s == '2*x'
+    expect(e).to be_a SymEngine::Pow
```
Comment (abinashmeher999) [inline spec/arit_spec.rb]: Shouldn't this be `SymEngine::Mul` ?

File: spec/arit_spec.rb [hunk 2]
```diff
@@ -54,22 +54,32 @@
...
     assert(e.to_s == 'x + y') || 'y + x'
+    expect(e).to be_a SymEngine::Add
+
     e = x * y
     assert(e.to_s == 'x*y') || 'y*x'
+    expect(e).to be_a SymEngine::Mul
+
     e = Integer(2) * x
     assert e.to_s == '2*x'
+    expect(e).to be_a SymEngine::Pow
```
Comment (abinashmeher999) [inline spec/arit_spec.rb]: Same here.

**Comment by abinashmeher999** [comment]
I think we should at least stick to `be_an_instance_of`in the class specs because there we need to be sure that it is indeed an instance of the same class not an object where we are only enforcing the class to be somewhere in the inheritance tree. We can use `be_a`/`be_an` at the other specs like `arit_spec.rb`

File: spec/arit_spec.rb (hunk 1 above)
Comment (rajithv) [inline spec/arit_spec.rb]: @abinashmeher999 fixed this

**Comment by abinashmeher999** [comment]
@rajithv Just a suggestion. It is good to write the tests first. That way you are sure how you want your objects to behave in Ruby and not get influenced by the C like syntax. You can use `xit` to allow the tests to fail till the source is written.

**Comment by rajithv** [comment]
@abinashmeher999 sure.
I will do that in the future.
So I changed the checks to `be_an_instance_of` when the expression is required to exactly a `Rational` or an `Integer`, `Add` etc.

**Comment by abinashmeher999** [comment]
Looks good to me. However, you will have to rebase once https://github.com/symengine/symengine.rb/pull/15 is merged.

The binary and unary operators were tested in `basic_spec.rb` because `Add`, `Mul` and `Pow` weren't supported at that time. Once this PR is merged try to move those out into `add_spec`, `mul_spec` and `pow_spec`. Then the tests should check if the result is an instance of their respective classes.

File: spec/basic_spec.rb [hunk 3]
```diff
@@ -157,7 +157,7 @@
           z = SymEngine::Symbol.new('z')
           e = (x**y / z)
           f = e.free_symbols
-          expect(f).to be_an_instance_of Set
+          expect(f).to be_a Set
```
Comment (isuruf) [inline spec/basic_spec.rb]: This change is unnecessary

File: spec/basic_spec.rb [hunk 4]
```diff
@@ -142,7 +142,7 @@
           z = SymEngine::Symbol.new('z')
           e = (x**y + z)
           f = e.args
-          expect(f).to be_an_instance_of Array
+          expect(f).to be_an Array
```
Comment (isuruf) [inline spec/basic_spec.rb]: This change is unnecessary

**Comment by isuruf** [comment]
+1 to merge

**Comment by certik** [comment]
+1 as well. Should we wait for #15 to be merged first?

**Comment by abinashmeher999** [comment]
We should have the tests run on this once which would be after #15.

**Comment by isuruf** [comment]
@rajithv, thanks for the PR

**Pull request merged**